#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Fix __bind redefinition problem
# DP: Related bugs: 
# DP: Dpatch author: Clint Adams
# DP: Patch author: Clint Adams
# DP: Upstream status: Not Submitted
# DP: Status Details:
# DP: Date: 2005-12-10

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.

--- glibc-2.3.6/sysdeps/unix/sysv/linux/sparc/sparc32/socket.S.old	2003-08-31 13:23:11.000000000 -0400
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/sparc/sparc32/socket.S	2005-12-10 12:46:23.313930095 -0500
@@ -40,7 +40,11 @@
    The .S files for the other calls just #define socket and #include this.  */
 
 #ifndef __socket
-#define __socket P(__,socket)
+# ifndef NO_WEAK_ALIAS
+#  define __socket P(__,socket)
+# else
+#  define __socket socket
+# endif
 #endif
 
 .globl __socket
@@ -105,4 +109,6 @@
 
 END (__socket)
 
+#ifndef NO_WEAK_ALIAS
 weak_alias (__socket, socket)
+#endif
--- glibc-2.3.6/sysdeps/unix/sysv/linux/sparc/sparc64/socket.S.old	2003-08-31 13:23:11.000000000 -0400
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/sparc/sparc64/socket.S	2005-12-10 12:46:23.313930095 -0500
@@ -40,7 +40,11 @@
    The .S files for the other calls just #define socket and #include this.  */
 
 #ifndef __socket
-#define __socket P(__,socket)
+# ifndef NO_WEAK_ALIAS
+#  define __socket P(__,socket)
+# else
+#  define __socket socket
+# endif
 #endif
 
 .globl __socket
@@ -105,4 +109,6 @@
 
 END (__socket)
 
+#ifndef NO_WEAK_ALIAS
 weak_alias (__socket, socket)
+#endif
